code/script.js

```diff
@@ -222,104 +227,189 @@ const generateBoard = () => {
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      alert(`Yes, the person has ${value} hair! Keep all people that has ${value} hair.`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value) 
+    }  
+    else {
+      alert(`No, the person doesn't have ${value} hair! Remove all people that have ${value} hair.`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  }
+  
+  else if (category === 'eyes') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`Yes, the person has ${value} eyes! Keep all people that have ${value} eyes.`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } 
+    else {
+      alert(`No, the person doesn't have ${value} eyes! Remove all people that have ${value} eyes.`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
+  else if (category === 'accessories') {
+    if (keep) {
+      alert(`Yes, the person wears ${value}! Keep all people that wears ${value}`)
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    } 
+    
+    else {
+      alert(`No, the person doesn't wear ${value}! Remove all people that wears ${value}`)
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+    }
+  } 
+  
+  else if (category === 'other') {
+    if (keep) {
+      alert(`Yes, the person has a ${value}! Keep all people that have a ${value}`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    } 
+    
+    else {
+      alert(`No, the person doesn't have a ${value}! Remove all people that have a ${value}`)
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+  } 
+  console.log(charactersInPlay)
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// Function: the user confirms his/her guess and the function checkMyGuess is invoked. 
```

Comment:
This function is missing the control for when a user cancels their guess, it still proceeds to the revealing the secret person.

Reply:
I totally forgot about making something happen when the user presses "cancel". Thank you for informning me of this! ☺️